docs/hack_loki_operator.md

```diff
@@ -0,0 +1,222 @@
+
+  If you don't want `lokistack-gateway` component then you can skip it by removing the `--with-lokistack-gateway` args from the `controller-manager` deployment:
+
+  ```console
+  kubectl edit deployment/controller-manager
+  ```
+
+  Delete the flag `--with-lokistack-gateway` from the `args` section and save the file. This will update the deployment and now you can create LokiStack instance using:
+
+  ```console
+  kubectl apply -f hack/lokistack_dev.yaml
+  ```
+
+  This will create `distributor`, `compactor`, `ingester`, `querier` and `query-frontend` components only.
+
+### Cleanup
+
+To cleanup deployments of the operator, you can use:
+
+```console
+make undeploy
+```
+
+It will undeploy controller from the configured Kubernetes cluster in ~/.kube/config
+
+## Hacking on Loki Operator on OpenShift
+
+### Requirements
+
+* Install [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) or [Openshift CLI](https://docs.openshift.com/container-platform/latest/cli_reference/openshift_cli/getting-started-cli.html) for communicating with the cluster. The guide below will be using `kubectl` for the same.
+* Create a running OpenShift cluster on AWS.
+* A container registry that you and your OpenShift cluster can reach. We recommend  [quay.io](https://quay.io/signin/).
+* Create an [S3 bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/create-bucket-overview.html) in one of the AWS Regions.
+
+### Installation of Loki Operator
+
+* Build and push the container image and then deploy the operator with:
+
+  ```console
+  make olm-deploy REGISTRY_ORG=$YOUR_QUAY_ORG VERSION=$VERSION
+  ```
+
+  where `$YOUR_QUAY_ORG` is your personal [quay.io](http://quay.io/) account where you can push container images and `$VERSION` can be any random version number such as `v0.0.1`.
+
+  The above command will deploy the operator to your active Openshift cluster defined by your local [kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/). The operator will be running in the `openshift-logging` namespace.
+
+* You can confirm that the operator is up and running using:
+
+  ```console
+  kubectl -n openshift-logging get pods
+  ```
+
+* Now you need to create a storage secret for the operator. This can be done using:
+
+  ```
+  make olm-deploy-example-storage-secret
+  ```
+
+  OR
+
+  ```console
+  ./hack/deploy-example-secret.sh openshift-logging
+  ```
+
+  This secret will be available in openshift-logging namespace. You can check the `hack/deploy-example-secret.sh` file to check the content of the secret.
+
+* Once the object storage secret is created, you can now create a LokiStack instance to get the various components of Loki up and running:
+
+  ```console
```

Comment:
I went through these instructions and had to add `-n openshift-logging` to make them work.

Reply:
Good catch!